vanessaspear/passport-api
# Updated the packing model and view to include a "packed" field

## Description 
The client needs to be able to display whether or not an item has been packed.  The client uses checkboxes for this functionality.  The server and database need to be updated to support the packing list checkboxes and a boolean "packed" field has been added to correspond with the client checkboxes.  The default state of "packed" is false, and when the user clicks the checkbox, the client will send a request to update the "packed" field to true. 

## Changes

- Updated the packing list model to have a boolean packed field to allow the user to set whether or not an item has been packed
- Update the associated serializer in the the packing view to include the packed field
- Update the packing list fixture objects with the packed field

## Requests / Responses

**Request**

POST `/packinglist` Creates a new item

```json
{
    "trip_id": 1,
    "item": "Flip-flops",
    "packed": false
}
```

**Response**

HTTP/1.1 201 OK

```json
{
    "id": 20,
    "trip": {
        "id": 1,
        "name": "Spring Break Trip",
        "city": "Key West",
        "state_or_country": "Florida",
        "departure_date": "2018-03-16",
        "return_date": "2018-03-18",
        "user": 1,
        "reasons": [
            1
        ]
    },
    "item": "Flip-flops",
    "packed": false
}
```

## Testing

```
git fetch origin vs-packing-checkboxes
git checkout vs-packing-checkboxes
```
- Delete database and run migrations to update packing database models
- Load fixtures 
- Pull down client updates from passport-client repo, pull request # 27
- Add an item to packing list 
- Select the checkbox and verify it shows as checked 
- Refresh the page and verify it still shows as checked
- Delete an item and verify it has been removed from the packing list
